### PR TITLE
Additional createPlaylist parameters

### PR DESCRIPTION
**Pull request by doodlezucc**
This PR adds optional parameters `public`, `collaborative` and `description` (as stated in the API reference) to the createPlaylist function.

For testing purposes, I wrote an example call in example_auth.dart. You can uncomment [this line](https://github.com/rinukkusu/spotify-dart/compare/master...doodlezucc:master#diff-453b7c78bb316aac18630cae4a77d3edfb33bbd8912bf6eb65f53c1ee7d1da94R20) to try it out.

### CODE REVIEW

File: lib/src/endpoints/playlists.dart [hunk 1]
```diff
@@ -36,10 +36,16 @@ class Playlists extends EndpointPaging {
   /// [userId] - the Spotify user ID
   ///
   /// [playlistName] - the name of the new playlist
```
Comment (rinukkusu) [inline lib/src/endpoints/playlists.dart:38]: Please document the other arguments as well. 😃

File: example/example_auth.dart [hunk 2]
```diff
@@ -64,7 +65,21 @@ void _devices(SpotifyApi spotify) async =>
...
+    'Cool New Playlist 2',
+    description: 'Songs to test by',
+    public: false,
+  )
+      .then((playlist) {
+    print('Private playlist created!');
+  }).catchError(_prettyPrintError);
+}
+
+dynamic _prettyPrintError(Object error) {
```
Comment (hayribakici) [inline example/example_auth.dart:82]: Do you need the `dynamic` here?😆

File: lib/src/endpoints/playlists.dart (hunk 1 above)
Comment (doodlezucc) [reply lib/src/endpoints/playlists.dart:38]: Done!

Comment (doodlezucc) [reply lib/src/endpoints/playlists.dart:38]: There are some inconsistencies in this repository when it comes to comments though. For example, the capitalization of the first letter varies, sometimes in [a single doc comment](https://github.com/doodlezucc/spotify-dart/blob/master/lib/src/endpoints/playlists.dart#L118-L121). The added comments should fit in.

File: example/example_auth.dart (hunk 2 above)
Comment (doodlezucc) [reply example/example_auth.dart]: Sorry, you're right, dynamic is not needed. But I did have to change `Exception` to `Object` for catchError to work.

File: lib/src/endpoints/playlists.dart (hunk 1 above)
Comment (rinukkusu) [reply lib/src/endpoints/playlists.dart:38]: Yeah, that's true - but let's pick it up as we go. Thanks for your engagement!! 😄

**Review by rinukkusu — APPROVED** [review]
LGTM

**Pull request merged**